lightning/src/routing/scoring.rs

```diff
@@ -1195,8 +1201,10 @@ DirectedChannelLiquidity< L, HT, T> {
 				// If we don't have any valid points (or, once decayed, we have less than a full
 				// point), redo the non-historical calculation with no liquidity bounds tracked and
 				// the historical penalty multipliers.
-				let (numerator, denominator) = success_probability(amount_msat, 0,
-					available_capacity, available_capacity, score_params, true);
+				let (numerator, denominator) = success_probability(
+					total_inflight_amount_msat, 0, available_capacity, available_capacity,
+					score_params, true,
+				);
 				let negative_log10_times_2048 =
 					log_approx::negative_log10_times_2048(numerator, denominator);
 				res = res.saturating_add(Self::combined_penalty_msat(amount_msat, negative_log10_times_2048,
```

Comment:
IIUC, we want to use the total amount when computing the success probability since in-flight HTLCs will affect this. However, when computing the actual penalties we use the payment amount (i.e., `amount_msat`) *and* the success probability. So each successive HTLC in-flight would cause the penalty to increase since the success probability would decrease (assuming the total amount is above the minimum liquidity estimate).

That said, as written, our docs may need to be clearer about what amount is used to compute the success probability (total amount) and what amount is used for the penalty (payment amount w/ success probability).

Reply:
> That said, as written, our docs may need to be clearer about what amount is used to compute the success probability (total amount) and what amount is used for the penalty (payment amount w/ success probability).

Updated the docs in the latest push, BTW.